pilot/pkg/config/clusterregistry/conversion.go

```diff
@@ -0,0 +1,219 @@
+/*
+Copyright 2018 Istio Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package clusterregistry
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"reflect"
+	"strconv"
+	"strings"
+
+	"github.com/golang/glog"
+	multierror "github.com/hashicorp/go-multierror"
+	"go.uber.org/multierr"
+
+	"k8s.io/apimachinery/pkg/util/yaml"
+
+	"istio.io/istio/pilot/pkg/serviceregistry"
+)
+
+var (
+	supportedExtensions = map[string]bool{
+		".yaml": true,
+		".yml":  true,
+	}
+)
+
+// annotations for a Cluster
```

Comment:
More comments and examples ? What are the values of each annotation, where are they set, etc.

Reply:
I was thinking to do this, but didn't find a right place to do so yet. Any suggestions?

Reply:
If I knew what they did.... That's why I asked for comments...